components/notebook-controller/controllers/notebook_controller.go

```diff
@@ -528,9 +529,10 @@ func (r *NotebookReconciler) SetupWithManager(mgr ctrl.Manager) error {
 
 	eventToRequest := handler.ToRequestsFunc(
 		func(a handler.MapObject) []ctrl.Request {
+			event := a.Object.(*v1.Event)
 			return []reconcile.Request{
 				{NamespacedName: types.NamespacedName{
-					Name:      a.Meta.GetName(),
+					Name:      nbNameFromInvolvedObject(&event.InvolvedObject),
```

Comment:
I am not sure about this change. Later, in the reconcile call, we need to fetch the actual event:

```
getEventErr = r.Get(ctx, req.NamespacedName, event)
```

Reply:
Oh wow, I hadn't seen that because I never expected to see two kinds inside the same reconcile queue.
This seems like a very bad practice.

Thanks for pointing it out, with the new handling we never enqueue events.

Reply:
I don't think that code is obsolete. The goal of enqueueing events that were originated in a pod or sts related to a notebook is:
 
* Fetch the event.
* Fetch the notebook related to the previous event (using `nbNameFromInvolvedObject`).
* Reissue the event as if it was the notebook the source of the event.

That code was introduced to solve https://github.com/kubeflow/kubeflow/issues/3772 and it is used by the jupyte web-app to enrich the information shown to the user (see [this](https://github.com/kubeflow/kubeflow/pull/4214/files#diff-e737059d17e1c37ba095426315b34847R110)).

So I guess the only change needed in this PR is to fix `nbNameFromInvolvedObject`, right?

Reply:
You're right, the reissuing is the only thing that's not handled.
I'm not very fond of this practice because it can lead to name collisions (event name == sts name) and skipped reconciliations.
They should be very rare in practice though.
I'm ok with leaving it like this for now.